api/v1alpha3/types.go

```diff
@@ -132,6 +132,7 @@ type Instance struct {
 	FailureDomain  string            `json:"failureDomain,omitempty"`
 	SecurityGroups *[]string         `json:"securigyGroups,omitempty"`
 	Networks       *[]Network        `json:"networks,omitempty"`
+	AccessSubnet   string            `json:"accessSubnet,omitempty"`
```

Comment:
hm does it make sense to call the property in instance differently than the one in the spec?

Reply:
Hm maybe I missed something / or read it incorrectly. Could it be that this adds the Subnet property to the OpenStackMachine and the AccessSubnet to the Instance struct which seems to be only used for the status of the Bastion host?

Reply:
`Instance` struct seems to be used as a status of bastion and as a holder of state of a created VM, but at the same time, it is used as an input for `createInstance`. I don't know why this is so, but I wanted to minimize the impact of this PR.
There are basically two possibilities how to forward the subnet UUID into the `createInstance` function:
 - add this property to this `Instance` struct (what I did)
 - add another (optional) parameter to `createInstance` function
The second option seemed inappropriate, taking into consideration the current signature:
```go
func createInstance(is *Service, clusterName string, i *infrav1.Instance) (*infrav1.Instance, error)
```
Adding `accessSubnetUUID string` does not feel right, but now that I'm thinking about it, the advantage would be that it would not change the public contract in bastion host status.

Just let me know, I can add another commit and forward the subnet UUID in an additional argument.

It also might make sense to keep it as it is and add the same possibility to specify subnet UUID for bastion host as well (adding similar `Subnet` property to `Bastion` struct https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/437e82faec1f9b77493e8e2bc387ea704c9d2373/api/v1alpha3/types.go#L251-L266)
Then having this property in Bastion status starts making sense.